misc/efi-stub/README.rst

```diff
@@ -0,0 +1,120 @@
+
+UEFI BIOS/shim -> acrn.efi -> ACRN HV -> Service VM (pre-launched VM in parallel)
+
+1. UEFI BIOS/shim verifies & loads a signed acrn.efi
+2. acrn.efi unpacks ACRN HV image and VM kernels from a stitched container blob
+3. acrn.efi loads ACRN HV to memory
+4. acrn.efi prepares MBI to stores Service VM & pre-launched VM kernel info
+5. acrn.efi hands over control to ACRN HV with MBI
+6. ACRN HV boots Service VM (and pre-launched VM in parallel)
+
+As the container blob format, ``ACRN v2.5 EFI-Stub`` uses the `Slim Bootloader
+Container Boot Image <https://slimbootloader.github.io/how-tos/create-container-boot-image.html>`_.
+
+Verified Configurations
+***********************
+- ACRN Hypervisor Release Version 2.5
+- hybrid_rt scenario
+- TGL-U RVP board
+- CONFIG_MULTIBOOT2=y (as default)
+- CONFIG_RELOC=y (as default)
+
+
+Building
+********
+
+Build Dependencies
+==================
+
+- Build Tools and Dependencies described in the `Getting Started Guide <https://projectacrn.github.io/latest/getting-started/building-from-source.html>`_
+- gnu-efi package
+- Service VM Kernel bzImage
+- pre-launched RTVM Kernel bzImage
+- `Slim Bootloader Container Tool <https://slimbootloader.github.io/how-tos/create-container-boot-image.html>`_
+
+The Slim Bootloader Tools can be downloaded from its github `project <https://github.com/slimbootloader/slimbootloader>`_.
+The verified version is the commit `9f146afd47 <https://github.com/slimbootloader/slimbootloader/tree/9f146afd47e0ca204521826a583d55388850b216>`_.
+You may use the `meta-acrn <https://github.com/intel/meta-acrn>`_ to build Service VM Kernel and pre-launched one.
+
+Build EFI-Stub for TGL-U RVP hybrid_rt
+======================================
+
+.. code-block:: none
+
+   $ cd acrn-hypervisor
+   $ make BOARD=tgl-rvp SCENARIO=hybrid_rt hypervisor
+   $ make BOARD=tgl-rvp SCENARIO=hybrid_rt -C misc/efi-stub/ \
+     HV_OBJDIR=`pwd`/build/hypervisor/ EFI_OBJDIR=`pwd`/build/hypervisor/misc/efi-stub `pwd`/build/hypervisor/misc/efi-stub/boot.efi
+
+Create Container
+================
+
+.. code-block:: none
+
+   $ touch hv_cmdline.txt
+   $ echo RT_bzImage    > vm0_tag.txt
+   $ echo Linux_bzImage > vm1_tag.txt
+   $ echo ACPI_VM0      > acpi_vm0.txt
+
+   $ python3 GenContainer.py create -cl \
+     CMDL:./hv_cmdline.txt \
+     ACRN:build/hypervisor/acrn.32.out \
+     MOD0:./vm0_tag.txt \
+     MOD1:./vm0_kernel \
+     MOD2:./vm1_tag.txt \
+     MOD3:./vm1_kernel \
+     MOD4:./acpi_vm0.txt \
+     MOD5:build/hypervisor/acpi/ACPI_VM0.bin \
+     -o sbl_os \
+     -t MULTIBOOT \
+     -a NONE
+
+You may optionally put HV boot options to the hv_cmdline.txt. The vm0_kernel is the Kernel bzImage of the pre-launched RTVM,
+and the vm1_kernel is the one of the Service VM in the above case.
+
+Stitch Container to EFI-Stub
+============================
+
+.. code-block:: none
+
+   $ objcopy --add-section .hv=sbl_os --change-section-vma .hv=0x6e000 --set-section-flags .hv=alloc,data,contents,load \
```

Comment:
I did not expect users themselves execute these sort of commands. Instead my initial wish was meta-acrn supports a recipe to build acrn.efi so that users don't need to understand all background commands flow. Nevertheless I am fine to revise this README file to run these commands outside of the acrn-hypervisor directory.

Reply:
fixed, thank you